cmake/developer_package/target_flags.cmake

```diff
@@ -35,7 +35,7 @@ if(MSVC64 OR MINGW64)
   _ie_process_msvc_generator_platform(X86_64)
 elseif(MINGW OR (MSVC AND NOT CMAKE_CROSSCOMPILING))
   _ie_process_msvc_generator_platform(X86)
-elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*" OR (APPLE AND CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64"))
```

Comment:
BTW, what is the system processor if you pass `-DCMAKE_OSX_ARCHITECTURES=x86_64`? I expect it to be something like `X86_64 `

Reply:
we can remove `APPLE AND`

Reply:
For `X86_64` I get the next error:
```
clang: error: invalid arch name '-arch X86_64'
```